### ISSUE TITLE
cannot set ApplicationVersion globally in web projects?

### ISSUE THREAD

**Issue by skoub** [MindscapeHQ/raygun4net#216]
hi!

in the documentation, it says that the RayGun4Net will try to set ApplicationVersion automatically. But in my MVC project, it is always "Not supplied". This is normal because in a web project, `Assembly.GetEntryAssembly()` always return null:

```
public IRaygunMessageBuilder SetVersion(string version)
{
    if (!String.IsNullOrEmpty(version))
    {
    _raygunMessage.Details.Version = version;
    }
    else
    {
        var entryAssembly = Assembly.GetEntryAssembly();
        if (entryAssembly != null)
        {
            _raygunMessage.Details.Version = entryAssembly.GetName().Version.ToString();
        }
        else
        {
            _raygunMessage.Details.Version = "Not supplied";
        }
    }
    return this;
}
```

So because of this, i would like to set it programmatically in my Application_Start but it seems that this property cannot be set globally. 

Am I missing something? or is there any plan to implement this feature so we can set a few things one time at the Application startup?

**Comment by jamiepenney** [comment]
We call the SetVersion() method on the RaygunClient after we create it. We've got a helper method which creates a new RaygunClient, then sets the version and user on it. 

````csharp
    public static RaygunClient GetRaygunClient()
    {
      return (RaygunClient) AddVersionToRaygunClient(AddUserToRaygunClient(new RaygunClient()));
    }

    private static RaygunClientBase AddVersionToRaygunClient(RaygunClientBase client)
    {
      try
      {
        // Attempt to set the version if we can
        client.ApplicationVersion = Assembly.GetExecutingAssembly().GetName().Version.ToString();
      }
      // ReSharper disable once EmptyGeneralCatchClause
      catch { }

      return client;
    }

    private static RaygunClientBase AddUserToRaygunClient(RaygunClientBase client)
    {
      if (Thread.CurrentPrincipal != null)
      {
        client.UserInfo = GetRaygunIdentifierMessage(Thread.CurrentPrincipal.RaygunIdentity());
      }
      return client;
    }
````

This is in a class in our Website project, so `GetExecutingAssembly()` returns the website assembly.

**Comment by skoub** [comment]
I think the doc should be corrected and say that in a web application, the Version cannot be set automatically and put a sample code to show how to do it manually.

If you want, i can work something on my side to let user set the ApplicationVersion the same way you can set the ApiEndpoint for example: `RaygunSettings.Settings.ApplicationVersion` and push it back to you.

**Comment by QuantumNightmare** [comment]
Hi skoub,

apologies for the confusion, the documentation is currently in the process of being updated. As Jamie mentioned, you will be able to set the version in a single place in your application by setting the ApplicationVersion property of the RaygunClient instance that is used to report your exceptions.

If you are using the Raygun HTTP module, then you may notice that the module uses its own RaygunClient instance to report exceptions. You can provide you own RaygunClient instance to the Http module by going to the docs here: https://raygun.io/docs/languages/net/asp and follow the "Providing a custom RaygunClient to the HTTP module" instructions.

Either follow the above instructions, or feel free to implement the application-version in the RaygunSettings as you mentioned which would be more convenient. We'd be happy to get a PR for that change if you do take that route.

Please let us know if you have further questions about this.

**Comment by skoub** [comment]
I will implement the ApplicationVersion in the RaygunSettings. I think it's the way to go. I will let the possibility to set the application version at the RaygunClient instance to override the RaygunSettings. Is that ok with you?

**Comment by QuantumNightmare** [comment]
Thanks skoub,

Yes that sounds good, allowing the RaygunClient property value to override the RaygunSettings property value would be the best behavior.

**Comment by skoub** [comment]
i've sent a PR #217

**Comment by QuantumNightmare** [comment]
Thanks for the PR. This has now been merged. I'll reply back when this has been published to NuGet.

**Comment by skoub** [comment]
no problem!

**Comment by QuantumNightmare** [comment]
Your changes are now available in the latest version on NuGet! (v4.2.0)
